File: cpp/3-5.cpp
#include "disp.h"//使用Display单元库
pbitmap img;
int main(){
createwin(800,600);//建立窗口
img=loadbmp("display.png");//读取图片
drawbmp(img,(getwidth()-img->width*2)/2,(getheight()-img->height*2)/2,img->width*2,img->height*2);//绘制拉伸图片
freshwin();//刷新窗口
msgbox("绘制拉伸完成");//输出绘制完成信息
drawbmp(img,img->width/4,img->height/4,img->width/2,img->height/2,(getwidth()-img->width*2)/2,(getheight()-img->height*2)/2,img->width*2,img->height*2);//绘制剪切拉伸图片
freshwin();//刷新窗口
msgbox("绘制剪切完成");//输出绘制完成信息
return 0;}

File: pas/3-5.pas
uses Display;//使用Display单元库
var img:pbitmap;
begin
CreateWin(800,600);//建立窗口
img:=LoadBMP('display.png');//读取图片
DrawBMP(img,(GetWidth()-img^.Width*2)div 2,(GetHeight()-img^.Height*2)div 2,img^.Width*2,img^.Height*2);//绘制拉伸图片
FreshWin();//刷新窗口
Msgbox('绘制拉伸完成');//输出绘制完成信息
DrawBMP(img,img^.Width div 4,img^.Height div 4,img^.Width div 2,img^.Height div 2,(GetWidth()-img^.Width*2)div 2,(GetHeight()-img^.Height*2)div 2,img^.Width*2,img^.Height*2);//绘制剪切拉伸图片
FreshWin();//刷新窗口
Msgbox('绘制剪切完成');//输出绘制完成信息
end.
